### ISSUE TITLE
Can you switch some letters in Ukrainian? They r didnt use so far as others. Pleeeeease

### ISSUE THREAD

**Issue by UkrainianFrog** [sspanak/tt9#678]
- [а, б, в, г] # 2
  - [д, е, ж, з] # 3
  - [и, і, к, л] # 4
  - [м, н, о, п] # 5
  - [р, с, т, у] # 6
  - [ф, х, ц, ч] # 7
  - [ш, щ, й, ґ,] # 8
  - [ь, є, ю, я, ї] # 9

**Comment by sspanak** [comment]
What is the benefit of doing this? 

When using Predictive mode, it doesn't matter how many letters are there on a key, because you need to press each key only once. And in АБВ, you can scroll in both directions, so all letters are at most 2-3 clicks away. You can't reduce the amount of required clicks this way.

Moreover, the current layout is logically organized, i.e. all Г-like, E-like and И-like letters are on their own key, so you don't have to think too much when you want to find them. Relocating them on some random keys will confuse many users.

In summary, I must refuse this request.

Nevertheless, this is an open-source app. You are free to modify the layout as per your liking and build an APK for personal use.

**Comment by UkrainianFrog** [comment]
We dont use  ґ ї є so often, so its take forever to type basic words, bc  "i к л д е ж з" are  to far (4-5 clicks), but even more rarely usable ш щ has only 1-2 clicks.  It`s just so uncomfortamle to type! I cant use predictive mode bc there is no cellphone in the world with ukrainian letters on the buttons (physical keyboard).  

Thanks for your reply, I`ll learn how to modify it for personal use. Greetings from Ukraine!

**Comment by sspanak** [comment]
> there is no cellphone in the world with ukrainian letters on the buttons (physical keyboard)

This true for many languages, yet people find a way to manage it. This includes myself. I have also learned to type in Bulgarian, even though there is no phone in the world which has the layout printed on.

I would really recommend investing some time and learning the layout. It will pay back in much, much faster typing in the future. Multi-click is uncomfortable no matter what you do. 

To make the process of learning easier, you may want to enable the on-screen keys, so that you can look at them all the time. Go to Settings -> Appearance and change "On-screen Layout" to "Virtual Numpad". Also, you may want to reduce the key height to minimum, to save some screen space. If your phone has a touchscreen, you can also double tap or slide up/slide down the Status Bar to expand or minimize the virtual keys. The Status Bar is where the language name and the suggestions are displayed. 

This has helped me while I was working on languages that I can not read, like Arabic, Hebrew, Hindi or Korean.

If you still insist on building your own APK with your custom layout, you will need Android Studio. Just import the project from GitHub and follow the [development instructions](https://github.com/sspanak/tt9/blob/master/CONTRIBUTING.md).
